src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -571,13 +580,17 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     const textTracks = this._videoElement.textTracks;
     if ((textTrack instanceof PKTextTrack)
       && (textTrack.kind === 'subtitles' || textTrack.kind === 'captions')
-      && textTracks && textTracks[textTrack.index]) {
+      && textTracks) {
       this._removeNativeTextTrackChangeListener();
+      let selectedTrack;
       this._disableTextTracks();
-      textTracks[textTrack.index].mode = 'hidden';
-      NativeAdapter._logger.debug('Text track changed', textTrack);
-      this._onTrackChanged(textTrack);
-      this._addNativeTextTrackChangeListener();
+      [selectedTrack] = Object.values(textTracks).filter(track=> track.language === textTrack.language);
```

Comment:
`const track = this._el.textTracks[textTrack.index];`

Reply:
we will rely on the language to get the selected track, because when adding an external track, we might override a dom track cues with new cues, and this will make the textTrack.index not relevant. there is no use case of two captions with the same lang, so im relying on that

we will